File: Frontend/angular-app/src/app/modules/admin/repositories/base-value/commands/update-base-value-command.ts
import {Inject} from "@angular/core";
import {ServiceResult} from "../../../../../core/models/service-result";
import {IRequest, IRequestHandler} from "../../../../../core/services/mediator/interfaces";
import {BaseValue} from "../../../entities/base-value";
import {HttpService} from "../../../../../core/services/http/http.service";
import {ApplicationError} from "../../../../../core/exceptions/application-error";
import {MediatorHandler} from "../../../../../core/services/mediator/decorator";
import {NotificationService} from "../../../../../shared/services/notification/notification.service";
import {ValidationRule} from "../../../../../core/validation/validation-rule";
import {HttpRequest} from "../../../../../core/services/http/http-request";

export class UpdateBaseValueCommand extends IRequest<UpdateBaseValueCommand, BaseValue> {
  public id: number | undefined = undefined;
  public parentId: number | undefined = undefined;
  public baseValueTypeId: number | undefined = undefined;
  public title: string | undefined = undefined;
  public value: string | undefined = undefined;
  public isReadOnly: boolean | undefined = undefined;
  public orderIndex: number | undefined = undefined;
  public uniqueName: string | undefined = undefined;
  public code: string | undefined = undefined;

  constructor() {
    super();
  }

  mapFrom(entity: BaseValue): UpdateBaseValueCommand {
    this.mapBasics(entity,this)
    return this;
  }

  mapTo(): BaseValue {
    throw new ApplicationError(UpdateBaseValueCommand.name, this.mapTo.name, 'Not Implemented Yet')
  }

  get url(): string {
    return "/admin/basevalue/update";
  }

  get validationRules(): ValidationRule[] {
    return [];
  }
}

@MediatorHandler(UpdateBaseValueCommandHandler.name)
export class UpdateBaseValueCommandHandler implements IRequestHandler<UpdateBaseValueCommand, BaseValue> {
  constructor(
    @Inject(HttpService) private _httpService: HttpService,
    @Inject(NotificationService) private _notificationService: NotificationService,
  ) {
  }

  async Handle(request: UpdateBaseValueCommand): Promise<BaseValue> {
    let httpRequest: HttpRequest<UpdateBaseValueCommand> = new HttpRequest<UpdateBaseValueCommand>(request.url, request);

    return await this._httpService.Put<UpdateBaseValueCommand, ServiceResult<BaseValue>>(httpRequest).toPromise().then(response => {
      this._notificationService.showSuccessMessage()
      return response.objResult
    })

  }
}

File: Backend/Microservices/Admin/Eefa.Admin.Application.Refactor/UseCases/BaseValue/Command/Update/UpdateBaseValueCommand.cs
using System.Threading;
using System.Threading.Tasks;
using AutoMapper;
using MediatR;

public class UpdateBaseValueCommand : IRequest<ServiceResult<BaseValueModel>>, IMapFrom<BaseValue>
{
    public int Id { get; set; }
    public int BaseValueTypeId { get; set; } = default!;
    public int? ParentId { get; set; }
    public string Title { get; set; } = default!;
    public string UniqueName { get; set; } = default!;
    public string Value { get; set; } = default!;
    public int OrderIndex { get; set; } = default!;
    public bool IsReadOnly { get; set; } = default!;
    public string Code { get; set; }


    public void Mapping(Profile profile)
    {
        profile.CreateMap<UpdateBaseValueCommand, BaseValue>()
            .IgnoreAllNonExisting();
    }
}


public class UpdateBaseValueCommandHandler : IRequestHandler<UpdateBaseValueCommand, ServiceResult<BaseValueModel>>
{
    private readonly IUnitOfWork _unitOfWork;
    private readonly IMapper _mapper;

    public UpdateBaseValueCommandHandler(IUnitOfWork unitOfWork, IMapper mapper)
    {
        _mapper = mapper;
        _unitOfWork = unitOfWork;
    }

    public async Task<ServiceResult<BaseValueModel>> Handle(UpdateBaseValueCommand request, CancellationToken cancellationToken)
    {
        BaseValue entity = await _unitOfWork.BaseValues.GetByIdAsync(request.Id);

        _mapper.Map(request, entity);

        _unitOfWork.BaseValues.Update(entity);
        await _unitOfWork.SaveChangesAsync(cancellationToken);
        
        return ServiceResult.Success(_mapper.Map<BaseValueModel>(entity));
    }
}